purview/utils/general.py

```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import re
+import logging
+from datetime import datetime
+
+log = logging.getLogger(__name__)
+
+
+def timestamp_to_datetime(timestamp):
+    """
+    Parses a timestamp and return a datetime object
+    :param timestamp: a string of a timestamp
+    :return: a datetime object
+    """
+    log.debug('Trying to parse the timestamp "{0}"'.format(timestamp))
+    combinations = (
+        (r'^(?P<datetime>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2})(?:\.\d+)?$', '%Y-%m-%d %H:%M:%S'),
+        (r'^(?P<datetime>\d{4}-\d{2}-\d{2})$', '%Y-%m-%d'),
```

Comment:
Just one concern, I don't think it will match something like 2016-3-3

Reply:
Good point. I'll tweak the regex.